### ISSUE TITLE
Geo-Distributed RedisGraph Replication support

### ISSUE THREAD

**Issue by BhanuPrakash531** [RedisGraph/RedisGraph#1026]
Hi as per redislabs, redis support geo distributed replication. Does it hold good for redis graph also?
https://docs.redislabs.com/latest/rs/concepts/intercluster-replication/

**Comment by swilly22** [comment]
Hi @BhanuPrakash531, unfortunately no, at the moment RedisGraph doesn't work with CRDB.

**Comment by BhanuPrakash531** [comment]
1. Is there any plan to have CRDB with RedisGraph in coming future?
2. Mean time, do you have any solution to carry out cross site replication across data centers to allow HA of redisgraph?

**Comment by swilly22** [comment]
1. I believe that's a difficult task, not sure when we'll address it
2. you can use replica-of https://docs.redislabs.com/latest/rs/administering/active-passive/
but please note that while performing the replication, memory consumption will double on the replica side.

**Comment by BhanuPrakash531** [comment]
Does this replica-of support for RedisGraph also?

**Comment by swilly22** [comment]
Yes, but please note the memory consumption during the replication phase.

**Comment by BhanuPrakash531** [comment]
Okay. Much Thanks for the information.

**Comment by BhanuPrakash531** [comment]
Hi @swilly22 , now that the issue Redis enterprise replica-of [#908](https://github.com/RedisGraph/RedisGraph/issues/908) is resolved, does this means the memory consumption during replication phase is improved?
Thanks in advance

**Comment by swilly22** [comment]
Yes it is.

**Comment by DvirDukhan** [comment]
@BhanuPrakash531 
The reduction in memory consumption is relevant for using RedisGraph in Redis 6.0.1 and up. In this case, during RDB save/load (BGSAVE and replication sequence) we split the graph into multiple keys for replication so the additional memory overhead of replicating a graph is significantly reduced. For 5.0.x we still pass the graph object as a while key.

**Comment by BhanuPrakash531** [comment]
Thanks for clarifying @swilly22 , @DvirDukhan . Cheers

**Comment by victorstewart** [comment]
So just to confirm. RedisGraph does NOT support Redis-Enterprise's Active-Active Replication feature where you have multiple masters, only master-slave replication patterns? And I guess this is because beyond the command is black-box to the replicator because whether the writes are dependent (and only the newest should reign) or independent (and they should all execute in time sequence) is unknown?

**Comment by BhanuPrakash531** [comment]
Hi @victorstewart , when will the Redis-Enterprise's Active-Active Replication feature be supported?
